int32 data
string name
int32[] data2
